deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -656,6 +661,37 @@ def clear_fpgas(self) -> None:
                 with StreamLogger('stdout'), StreamLogger('stderr'):
                     run(f"xbutil reset -d {card_bdf} --force")
 
+    def localize_xclbin(self, slotno: int) -> None:
+        """ download xclbin URI to remote node. """
+        assert slotno < len(self.parent_node.sim_slots)
```

Comment:
If you could provide a helpful assertion message, that might be useful.

Reply:
Sure.  I'll come up with something and add it to the other methods that make this same assertion without a message.

Reply:
I think this assertion should just be put into a function since this is done quite a bit.

Reply:
Or removed because letting Python raise IndexError is probably just as useful and reduces the amount of code in the manager.  I'll leave that to future cleanup PRs .

Reply:
Also, if we move it into another function, I wonder if mypy is smart enough to use the assert in it's type-checking.  I know this particular assert predates the type checking and this is more a curiosity that I'm voicing out-loud because we have quite a few mypy helper asserts that check for `None`ness of `Optional` types and I could envision asking the same thing about them.

Reply:
Mypy is smart enough to do this. I can do quick cleanup of this when I get the chance